lib/main/server.js

```diff
@@ -0,0 +1,58 @@
+var http = require('http');
+var url = require('url');
+var Scrapper = require('./scrapper.js');
+var conf = require('../../config.js');
+
+function Server() {
+  this.httpServer = http.createServer(this.processRequest.bind(this));
+}
+
+Server.prototype.start = function() {
+  var self = this;
+  self.httpServer.listen(conf.get('port'), conf.get('ip'), function(x){
+    console.log('running on http://' + self.httpServer.address().address + ":" + self.httpServer.address().port);
+  });
+}
+
+Server.prototype.processRequest = function(request, response) {
+  var self = this;
+  var urlParams = url.parse(request.url, true);
+  if (request.method === 'GET' && urlParams.pathname === '/I/want/title/') {
+    var addresses = (typeof(urlParams.query.address) === 'string') ? [urlParams.query.address] : urlParams.query.address;
+    var titles = [];
+    var body = [];
+
+    function processTitle(title, index){
+      titles.push(title);
+      if(index < addresses.length - 1){
+          console.log(this);
+          self.fetchTitle(addresses[index + 1], index + 1, processTitle);
+      } else {
+        response.statusCode = 200;
```

Comment:
As a general rule rendering should go out in a separate function to improve readability.